[libc++] Add visibility annotations to the std namespace with GCC (#133233)
llvm/llvm-project: libcxx/docs/ReleaseNotes/21.rst

- Updated formatting library to Unicode 16.0.0.

> - When using GCC, the ``std`` namespace is now annotated with ``[[gnu::visibility("default")]]``. This may cause changes

Reviewer: I would document that under "ABI Affecting Changes" instead.

> - When using GCC, the ``std`` namespace is now annotated with ``[[gnu::visibility("default")]]``. This may cause changes
>   in the symbols exported from shared libraries when building with ``-fvisibility=hidden``. This also fixes RTTI

Reviewer: ```suggestion
- When using GCC, the ``std`` namespace is now annotated with ``[[gnu::visibility("default")]]``. This may cause more
  symbols to be exported from shared libraries when building with ``-fvisibility=hidden``. This also fixes RTTI
```

>   comparison between shared libraries, since all RTTI has the correct visibility now.

Reviewer: ```suggestion
  comparison between shared libraries, since all RTTI has the correct visibility now. There is no behavior change on Clang.
```
